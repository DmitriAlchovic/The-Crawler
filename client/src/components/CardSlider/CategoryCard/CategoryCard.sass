.category-container
    height: 215px
    width: 12rem !important
    background-color: #f2fce4
    position: relative
    display: flex
    align-items: center
    justify-self: flex-start
    justify-content: center
    margin: 21px
    white-space: normal
    min-width: auto
    min-height: auto
    flex-shrink: 0

    .img-container
        width: 111px
        height: 111px
        position: relative
        display: flex
        align-items: center
        justify-content: center

        .category-img
            object-fit: cover
            width: inherit
            height: inherit

    .category-name
        font-size: 18px
        margin: 10px

    .category-items
        font-size: 14px
        font-weight: 500
        color: #adadad


.container-hidden
    display: none