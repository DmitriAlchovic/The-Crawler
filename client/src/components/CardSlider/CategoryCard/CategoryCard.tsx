import React from 'react';
import { Subcategory } from '../../../store/reducers/categorySlice';
import './CategoryCard.sass';

interface CategoryCardProps {
  category: Subcategory;
  index: number;
  sliderIndex: number;
  sliderSize: number;
}

export default function CategoryCard({
  category,
  index,
  sliderIndex,
  sliderSize,
}: CategoryCardProps) {
  return (
    <div
      className={
        index >= (sliderIndex - sliderSize) && index < sliderIndex ? 'category-container' : 'container-hiden'
      }
    >
      <div>
        <div className="img-container">
          <img className="category-img" alt="no img" src={category.image} />
        </div>
        <div className="category-name">{category.name}</div>
        <div className="category-items">{`items ${index}`}</div>
      </div>
    </div>
  );
}
