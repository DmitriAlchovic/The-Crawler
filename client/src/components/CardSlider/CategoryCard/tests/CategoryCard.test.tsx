import React from 'react';
import { render } from '@testing-library/react';
import * as reduxHooks from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import CategoryCard from '../CategoryCard';

jest.mock('react-redux');

describe('CategoryCard', () => {
  it('should create empty component', () => {
    jest.spyOn(reduxHooks, 'useSelector').mockReturnValue('');
    const component = render(
      <BrowserRouter>
        <CategoryCard
          category={{
            name: '',
            image: '',
            items: 0,
          }}
          index={0}
          sliderIndex={0}
          sliderSize={0}
        />
      </BrowserRouter>,
    );

    expect(component).toMatchSnapshot();
  });
});
