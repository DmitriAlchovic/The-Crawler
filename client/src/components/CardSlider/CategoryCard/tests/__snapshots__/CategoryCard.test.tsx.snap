// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard should create empty component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-hidden"
      >
        <div>
          <a
            href="/category//1"
          >
            <div
              class="img-container"
            >
              <img
                alt="no img"
                class="category-img"
                src=""
              />
            </div>
            <div
              class="category-name"
            />
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-hidden"
    >
      <div>
        <a
          href="/category//1"
        >
          <div
            class="img-container"
          >
            <img
              alt="no img"
              class="category-img"
              src=""
            />
          </div>
          <div
            class="category-name"
          />
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CategoryCardLoader should create empty component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-hidden"
      >
        <div>
          <a
            href="/category//1"
          >
            <div
              class="img-container"
            >
              <img
                alt="no img"
                class="category-img"
                src=""
              />
            </div>
            <div
              class="category-name"
            />
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-hidden"
    >
      <div>
        <a
          href="/category//1"
        >
          <div
            class="img-container"
          >
            <img
              alt="no img"
              class="category-img"
              src=""
            />
          </div>
          <div
            class="category-name"
          />
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
